### ISSUE TITLE
Remove promotion of toxic content?

### ISSUE THREAD

**Issue by zack6849** [twitter/the-algorithm#309]
Am I reading this right? 

https://github.com/twitter/the-algorithm/blob/main/home-mixer/server/src/main/scala/com/twitter/home_mixer/functional_component/decorator/HomeTweetTypePredicates.scala#L188

Are you actually pushing tweets to users /because/ they're detected as toxic?!

**Comment by VIEWVIEWVIEW** [comment]
No, you are not reading this right. Everything in that file is used for tagging tweets with certain features (has the tweet a link, is there anything toxic in it, has it x images etc). That code has nothing to do with actually *selecting* tweets for the home view.

The code in https://github.com/twitter/the-algorithm-ml/tree/main/projects/home/recap might help you if you are looking for this. However, you won't be able to replicate Twitter's feed 1:1 since some models and information are still missing.

**Comment by zack6849** [comment]
@VIEWVIEWVIEW thanks for clarifying, I still find it a bit concerning that toxicity is being fed into the ML stuff as an input, I suppose you could argue it may be being used to filter OUT overly toxic content depending on what it thinks I want to see, but I still find that a bit concerning.

**Comment by slweeb** [comment]
> I still find it a bit concerning that toxicity is being fed into the ML stuff as an input

how would the model learn about toxic stuff if it isn't fed toxic stuff

**Comment by kennethsequeira** [comment]
> how would the model learn about toxic stuff if it isn't fed toxic stuff

![image](https://user-images.githubusercontent.com/33246109/229264496-a4894b3f-7e50-42e2-8189-a5b55d109b3f.png)

**Comment by aloiscochard** [comment]
As the old saying goes "how would you know that you are dead if the machine already killed you?"

Anyway I think we are digressing a little bit here, we can close this issue as it was fixed already by https://github.com/twitter/the-algorithm/issues/309
